app/src/ui/toolbar/branch-dropdown.tsx

```diff
@@ -309,6 +309,9 @@ export class BranchDropdown extends React.Component<IBranchDropdownProps> {
       name: tip.branch.name,
       isLocal: tip.branch.type === BranchType.Local,
       onRenameBranch: this.onRenameBranch,
+      onViewBranchOnGitHub: tip.branch.upstreamRemoteName
+        ? this.onViewBranchOnGitHub
+        : undefined,
```

Comment:
This needs to check whether the repository is a GitHub repository (as opposed to a self-hosted, local, or a repository hosted on another Git provider).


```suggestion
      onViewBranchOnGitHub:
        isRepositoryWithGitHubRepository(this.props.repository) &&
        tip.branch.upstreamRemoteName
          ? this.onViewBranchOnGitHub
          : undefined,
```